sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -248,7 +248,18 @@ static az_result started(az_event_policy* me, az_event event)
     case AZ_MQTT5_EVENT_CONNECT_RSP:
       // There should be no connect response in this state as a connect request has not been sent or
       // a disconnect response has been received.
-      _az_RETURN_IF_FAILED(_az_hfsm_transition_peer((_az_hfsm*)me, started, faulted));
+      if (_az_hfsm_transition_peer((_az_hfsm*)me, started, faulted) != AZ_OK)
+      {
+        az_platform_critical_error();
+      }
+
+      ret = az_event_policy_send_inbound_event(
+          (az_event_policy*)this_policy,
+          (az_event){ .type = AZ_HFSM_EVENT_ERROR,
+                      .data = &(az_hfsm_event_data_error){ .error_type = event.type,
```

Comment:
The type of the event is `AZ_HFSM_EVENT_ERROR` but the internal error has the actual error, similar to what is done for _az_FAULTED_IF_FAILED

Reply:
but that information would be in .sender_event.type, right?

Reply:
Yes it would, the reasoning behind that is that the error sent inbound should be generic given that there are going to be state machines dependent on the connection so they should also know to go to faulted. They also may not know how to handle the error